frontend/src/components/mainPage/scene/subcomponents/NavigationPanel.vue

```diff
@@ -22,12 +22,18 @@
                     {
                         icon: "mdi-skip-backward",
                         action: () => {
+                            if (this.animationInterval) {
```

Comment:
Ten if powtarza się się za każdym razem jak jest wywoływana funkcja stopAnimation. Proponuję przenieść go do tej funkcji. I od razu proponuję zamiast pisać w tej funkcji `if (this.animationInterval) {coś tam coś tam}` to dać `if (!this.animationInterval) return;`. W ten sposób nie tworzymy niepotrzebnych poziomów zagnieżdżenia kodu.

Reply:
A i dla spójności dodaj wywołanie stopAnimation dla ikony mdi-skip-forward